primitives/staking/src/lib.rs

```diff
@@ -69,10 +75,62 @@ pub struct Stake<T: StakingInterface + ?Sized> {
 	/// This is only guaranteed to reflect the amount locked by the staking system. If there are
 	/// non-staking locks on the bonded pair's balance this amount is going to be larger in
 	/// reality.
-	pub total: T::Balance,
+	pub total: Balance,
 	/// The total amount of the stash's balance that will be at stake in any forthcoming
 	/// rounds.
-	pub active: T::Balance,
+	pub active: Balance,
+}
+
+/// A generic staking event listener.
+/// Note that the interface is designed in a way that the events are fired post-action, so any
+/// pre-action data that is needed needs to be passed to interface methods.
+/// The rest of the data can be retrieved by using `StakingInterface`.
+pub trait OnStakingUpdate<AccountId: Clone, Balance: Copy> {
```

Comment:
Same here, don't see a reason to bound anything to `Clone` within the trait def.

Reply:
I can't avoid bounding this, because `Stake` is a part of the signature and it requires those bounds. I'm not sure how else I can deriver`Clone` for `Stake`. And I need `Clone` for `Tuple` implementation.

Reply:
If you fix https://github.com/paritytech/substrate/pull/13079#discussion_r1112122310 `: Clone` should also be not needed here.

Reply:
As in, `sp-arithmetic` compiles fine with: 

```diff
diff --git a/primitives/staking/src/lib.rs b/primitives/staking/src/lib.rs
index a40ebecde39..9418095c8cc 100644
--- a/primitives/staking/src/lib.rs
+++ b/primitives/staking/src/lib.rs
@@ -63,7 +63,7 @@ impl<AccountId, Balance> OnStakerSlash<AccountId, Balance> for () {
 /// A struct that reflects stake that an account has in the staking system. Provides a set of
 /// methods to operate on it's properties. Aimed at making `StakingInterface` more concise.
 #[derive(Default, Clone, Debug, Eq, PartialEq)]
-pub struct Stake<AccountId: Clone, Balance: Clone> {
+pub struct Stake<AccountId, Balance> {
 	/// The stash account whose balance is actually locked and at stake.
 	pub stash: AccountId,
 	/// The total stake that `stash` has in the staking system. This includes the
@@ -85,7 +85,7 @@ pub struct Stake<AccountId: Clone, Balance: Clone> {
 /// Note that the interface is designed in a way that the events are fired post-action, so any
 /// pre-action data that is needed needs to be passed to interface methods.
 /// The rest of the data can be retrieved by using `StakingInterface`.
-pub trait OnStakingUpdate<AccountId: Clone, Balance: Copy> {
+pub trait OnStakingUpdate<AccountId, Balance> {
 	/// Fired when the stake amount of someone updates.
 	///
 	/// This is effectively any changes to the bond amount, such as bonding more funds, and

```

Reply:
All of that got fixed once I've removed the Tuple implementation and resorted to a default one, looking much better now.